clang/lib/CIR/CodeGen/CIRGenExpr.cpp

```diff
@@ -1472,9 +1472,10 @@ Address CIRGenFunction::emitArrayToPointerDecay(const Expr *e) {
   if (e->getType()->isVariableArrayType())
     return addr;
 
-  auto pointeeTy = mlir::cast<cir::ArrayType>(lvalueAddrTy.getPointee());
+  [[maybe_unused]] auto pointeeTy =
```

Comment:
These could be sunk into the asserts, but the expressions would need to be repeated. I was on the fence about which way to go.